Nicer console output while enhancing

vi opened the pull request:
file.png (tile    4 of    9)

instead of just printing dots

alexjc commented:
Quick review for this if you're keen to listen:
- Can you compute the `itertools.product` only once and put it in a list (DRY principle).
- You can determine its length using Python's `len` rather than a counter.
- Also, in the main loop you can use `enumerate` rather than another counter.

This should simplify the code and make it more Pythonic.  I'd welcome the change afterwards.

thomasbarwanitz commented:
Nice

thomasbarwanitz commented:
Nice

scribblemaniac commented:
Based on the changes in this PR, it seems like numtiles would just be equal to `math.ceil(original.shape[0] / s) * math.ceil(original.shape[1] / s)`, which would be faster and cleaner than `itertools.product`.

vi committed:
Nicer console output while enhancing

file.png (tile 4 of 9)

instead of just printing dots

vi commented:
* Rebased against master
* Implemented mentioned refactorings
* Printing numer of times before and after processing of each file
* Nicer output, usage of better working `\r` instead of hackier `\x7f`.

vi commented:
Can it process tiles in parallel BTW? Currently it looks like using only one core.

vi commented:
> I'd welcome the change afterwards.

@alexjc, So, is the change welcome now?

vi commented:
Rebased against v0.4
